tests/test_validation_java.py

```diff
@@ -20,7 +20,7 @@
     ENABLE_JAVA_TESTS = True
 except SystemError as ex:
     JAVA_CONFIGURED_REASON = f"{JAVA_CONFIGURED_REASON}: {str(ex)}"
-except JavaException as ex:
+except Exception as ex:
```

Comment:
Can you document why we need to catch raw `Exception` here? It leaves us open to unexpected behavior.

Reply:
I documented this in the commit message, but I'll put a comment in the code too.

Reply:
What exception is raised if `javac` is not found? `ImportError`? Can we catch that explicitly?

If we stick with catching `Exception`, then we should remove the `SystemError` clause above, since it's a subclass of `Exception` anyway. If we catch a more specific exception, we can still collapse these into a single line like:

```
except (JavaException, ImportError) as ex:
```

Reply:
It throws a JavaException on `from jnius import JavaException` if `javac` doesn't exist, so there's a bit of a chicken and egg problem going on there.

Reply:
Actually, I recalled this wrong, it's actually a subprocess exception. 

```
tests/test_validation_java.py:4: in <module>
    from jnius import JavaException
venv/lib/python3.8/site-packages/jnius/__init__.py:42: in <module>
    from .reflect import *  # noqa
venv/lib/python3.8/site-packages/jnius/reflect.py:17: in <module>
    class Class(with_metaclass(MetaJavaClass, JavaClass)):
venv/lib/python3.8/site-packages/six.py:856: in __new__
    return meta(name, resolved_bases, d)
jnius/jnius_export_class.pxi:114: in jnius.MetaJavaClass.__new__
    ???
jnius/jnius_export_class.pxi:164: in jnius.MetaJavaClass.resolve_class
    ???
jnius/jnius_env.pxi:11: in jnius.get_jnienv
    ???
jnius/jnius_jvm_dlopen.pxi:118: in jnius.get_platform_jnienv
    ???
jnius/jnius_jvm_dlopen.pxi:70: in jnius.create_jnienv
    ???
jnius/jnius_jvm_dlopen.pxi:47: in jnius.find_java_home
    ???
/usr/lib/python3.8/subprocess.py:411: in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
/usr/lib/python3.8/subprocess.py:512: in run
    raise CalledProcessError(retcode, process.args,
```

I'm going to collapse this down to a single exception though, since the error is logged and can be shown with `pytest -rs`:

```
SKIPPED [150] tests/test_validation_java.py:36: failed to configure java: Command '['which', 'javac']' returned non-zero exit status 1.
SKIPPED [52] tests/test_validation_java.py:46: failed to configure java: Command '['which', 'javac']' returned non-zero exit status 1.
```